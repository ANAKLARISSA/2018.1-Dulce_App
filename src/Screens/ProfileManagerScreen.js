import React from 'react';
import {View , Text} from 'react-native';
import AGRButton from '../Components/AGRButton';
import axios from 'axios';
import store from '../Reducers/store';
import {Container, Content, Spinner} from 'native-base';
import SmallLogo from '../Components/SmallLogo';
const styles = {
  container: {
    flex: 1,
    flexDirection: 'column',
    padding: 20,
    backgroundColor: '#FFF'

  },
  text: {
    fontSize: 20,
    marginBottom: 15,
    alignSelf: 'center'
  },
  name: {
    fontSize: 30,
    marginBottom: 15,
    alignSelf: 'center'
  },
  informacoes: {
    alignSelf: 'center'
  }
};

class ProfileManagerScreen extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      profile: [],
      loading: true
    };
  }

  componentDidMount() {
    this.setState({loading: true});

    const url = 'http://ec2-18-231-116-217.sa-east-1.compute.amazonaws.com/api/userManager/listById/?id=' + store.getState().currentUser.id;

    axios.get(url,{

      headers: {
        'x-access-token': store.getState().currentUser.token
      }

    })
      .then((response) => {this.setState({profile: response.data , loading: false});});
  }

  render() {
    return (
      <View style={styles.container}>
        {
          this.state.loading ? (
            <Container>
              <Content>
                <Spinner color='purple'/>
              </Content>
            </Container>
          ) : (
            <View style={styles.informacoes}>
            <SmallLogo />
              <Text style={styles.name}>{this.state.profile.name}</Text>
              <Text style={styles.text}>Matr√≠cula:{this.state.profile.registration}</Text>
              <Text style={styles.text}>Setor: {this.state.profile.sector}</Text>
              <Text style={styles.text}>Hospital: {this.state.profile.hospital}</Text>
              <View style={{marginTop: 60}} />
              <AGRButton text='Editar'onPress = {() => this.props.navigation.navigate('edit')}/>
            </View>
          )
        }
      </View>
    );
  }
}

export default ProfileManagerScreen;
