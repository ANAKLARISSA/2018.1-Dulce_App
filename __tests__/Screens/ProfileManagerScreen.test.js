import React from 'react';
import 'react-native';
import ProfileManagerScreen from '../../src/Screens/ProfileManagerScreen';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import expect from 'expect';

require('bezier');

Enzyme.configure({adapter: new Adapter()});

it('renders correctly', () => {
  const tree = renderer
  .create(<ProfileManagerScreen />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});

it('should test navigateToEditScreen correctly', () => {
  // This is to test component functions
  const navigation = {navigate: jest.fn()};
  let profileScreen = renderer
    .create(
      <ProfileManagerScreen
      navigation={navigation}
        dispatch={action =>
          expect(action)
          .toEqual(navigation.navigate)
        }
      />
    )
    .getInstance();

  profileScreen.navigateToEditScreen();
});


it('testing edit button',() => {
  
  const navigation  = {navigate: jest.fn()};
  const spy = jest.spyOn(ProfileManagerScreen.prototype, 'navigateToEditScreen');
  const wrapper = shallow(<ProfileManagerScreen navigation={navigation}/>);
  wrapper.update();
  wrapper.setState({loading: false});
  const edit = wrapper.find('IconButton').at(0);
  edit.simulate('press');
  expect(spy).toHaveBeenCalled();
  console.log(edit);
});

it('testing delete button',() => {
  
  const navigation  = {navigate: jest.fn()};
  const spy = jest.spyOn(ProfileManagerScreen.prototype, '_alert');
  const wrapper = shallow(<ProfileManagerScreen navigation={navigation}/>);
  wrapper.update();
  wrapper.setState({loading: false});
  const edit = wrapper.find('IconButton').at(1);
  edit.simulate('press');
  expect(spy).toHaveBeenCalled();
  console.log(edit);
});

const flushPromises = () => new Promise(resolve => setImmediate(resolve));

it('testing axios', async () => {
  const wrapper = shallow(<ProfileManagerScreen/>);
  await flushPromises();
  wrapper.update();
  expect(wrapper).toMatchSnapshot();
});