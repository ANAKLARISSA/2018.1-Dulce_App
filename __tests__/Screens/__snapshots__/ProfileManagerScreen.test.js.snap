// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F8F8F8",
            "borderBottomColor": "#a7a6ab",
            "borderBottomWidth": 0.5,
            "elevation": 3,
            "flexDirection": "row",
            "height": 64,
            "justifyContent": "center",
            "left": 0,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 15,
            "right": 0,
            "shadowColor": undefined,
            "shadowOffset": undefined,
            "shadowOpacity": undefined,
            "shadowRadius": undefined,
            "top": 0,
          },
          Object {
            "backgroundColor": "#5f4b8b",
          },
        ]
      }
      virtual={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 17,
                "fontWeight": "600",
                "textAlign": "center",
              },
              Object {
                "fontFamily": "Raleway",
              },
            ]
          }
          virtual={undefined}
        >
           
          Meu Perfil
           
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "height": 1334,
      }
    }
    virtual={undefined}
  >
    <RCTScrollView
      automaticallyAdjustContentInsets={false}
      contentContainerStyle={
        Array [
          Object {
            "padding": undefined,
          },
          undefined,
        ]
      }
      contentInset={
        Object {
          "bottom": 0,
        }
      }
      enableAutomaticScroll={true}
      enableResetScrollToCoords={true}
      extraHeight={75}
      extraScrollHeight={0}
      getScrollResponder={[Function]}
      handleOnScroll={[Function]}
      keyboardDismissMode="interactive"
      keyboardOpeningTime={250}
      keyboardShouldPersistTaps="handled"
      keyboardSpace={0}
      onScroll={[Function]}
      resetKeyboardSpace={[Function]}
      resetScrollToCoords={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      scrollEventThrottle={1}
      scrollForExtraHeightOnAndroid={[Function]}
      scrollToEnd={[Function]}
      scrollToFocusedInput={[Function]}
      scrollToPosition={[Function]}
      showsVerticalScrollIndicator={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        }
      }
      viewIsInsideTabBar={false}
      virtual={undefined}
    >
      <View>
        <ActivityIndicator
          animating={true}
          color="#5f4b8b"
          hidesWhenStopped={true}
          size="large"
          style={
            Object {
              "height": 80,
            }
          }
          virtual={undefined}
        />
      </View>
    </RCTScrollView>
  </View>
</View>
`;
